pulsar-broker/src/main/java/org/apache/pulsar/broker/lookup/TopicLookupBase.java

```diff
@@ -127,6 +127,7 @@ protected CompletableFuture<LookupData> internalLookupTopicAsync(TopicName topic
                                 log.debug("Lookup succeeded for topic {} -- broker: {}", topicName,
                                         result.getLookupData());
                             }
+                            pulsar().getBrokerService().getLookupRequestSemaphore().release();
```

Comment:
Can we move the semaphore release out of the `completeLookupResponseExceptionally`

```
    protected void completeLookupResponseExceptionally(AsyncResponse asyncResponse, Throwable t) {
        pulsar().getBrokerService().getLookupRequestSemaphore().release();
        Throwable cause = FutureUtil.unwrapCompletionException(t);
        asyncResponse.resume(cause);
    }
```

And make sure `internalLookupTopicAsync` will acquire the lock and release the lock property.
It's easier to read and maintain, reduces mistakes later.